src/screens/Developer/PaymentSettings/PaymentSettings.res

```diff
@@ -26,99 +26,284 @@ module InfoViewForWebhooks = {
+          </div>
+        </DesktopRow>
+        <RenderIf condition={showCloseIcon()}>
+          <div className="mt-6 flex gap-4">
+            <ModalCloseIcon onClick={_ev => removeInput(index, key)} />
+          </div>
+        </RenderIf>
       </div>
-    </DesktopRow>
+    }
   }
-}
-module WebHookAuthenticationHeaders = {
-  @react.component
-  let make = () => {
-    <div className="flex-1">
-      <p
-        className={`ml-4 text-fs-13 text-jp-gray-900 dark:text-jp-gray-text_darktheme dark:text-opacity-50 ml-1 !text-base !text-grey-700 font-semibold ml-1`}>
-        {"Custom HTTP Headers"->React.string}
-      </p>
-      <div className="grid grid-cols-5 flex gap-2">
-        {Array.fromInitializer(~length=4, i => i)
+  module WebHookAuthenticationHeaders = {
+    @react.component
+    let make = () => {
+      open LogicUtils
+      let form = ReactFinalForm.useForm()
+      let formState: ReactFinalForm.formState = ReactFinalForm.useFormState(
+        ReactFinalForm.useFormSubscription(["values"])->Nullable.make,
+      )
+      let (webHookAuthInputs, setWebHookAuthInputs) = React.useState(_ =>
+        Array.fromInitializer(~length=1, i => i)
+      )
+      React.useEffect(() => {
+        let length =
+          formState.values
+          ->getDictFromJsonObject
+          ->getDictfromDict("outgoing_webhook_custom_http_headers")
+          ->Dict.toArray
+          ->Array.length
+        let arr = Array.fromInitializer(~length=length > 0 ? length : 1, i => i)
+        setWebHookAuthInputs(_ => arr)
+        None
+      }, [])
+      let removeInput = (removeIndex, removeKey) => {
+        let details =
+          formState.values
+          ->getDictFromJsonObject
+          ->getDictfromDict("outgoing_webhook_custom_http_headers")
+          ->Dict.toArray
+          ->Array.filter(((key, _)) => key != removeKey)
+          ->Dict.fromArray
+        form.change("outgoing_webhook_custom_http_headers", details->Identity.genericTypeToJson)
+        setWebHookAuthInputs(_ =>
+          webHookAuthInputs->Array.filterWithIndex((_, index) => index != removeIndex)
+        )
+      }
+      let addInput = () => {
+        setWebHookAuthInputs(_ =>
+          Array.fromInitializer(~length=webHookAuthInputs->Array.length + 1, i => i)
+        )
+      }
+
+      let checkForEmpty = () => {
+        let arr =
+          formState.values
+          ->getDictFromJsonObject
+          ->getDictfromDict("outgoing_webhook_custom_http_headers")
+          ->Dict.toArray
+          ->Array.filter(((key, value)) =>
+            key->String.length != 0 && value->getStringFromJson("")->String.length != 0
+          )
+
+        arr->Array.length == webHookAuthInputs->Array.length ? true : false
+      }
+      <div className="flex-1">
+        <p
+          className={`ml-4 text-fs-13 text-jp-gray-900 dark:text-jp-gray-text_darktheme dark:text-opacity-50 ml-1 !text-base !text-grey-700 font-semibold ml-1`}>
```

Comment:
ml-x text-{color} is repeated property here, can you please remove that is not required?